source/plugins/freeze.plugin.ts

```diff
@@ -66,27 +73,31 @@ class FreezeTaw extends Plugin.ChatCommand
 			if (this.freeze)
 			{
 				// Block members from writing, and allow mods to write
+				await channel.updateOverwrite(message.guild.roles.everyone, {'SEND_MESSAGES': false}, "channel freeze");
 				await channel.updateOverwrite( memberRole, { 'SEND_MESSAGES': false }, "channel freeze" );
-				await channel.updateOverwrite( modRole, { 'SEND_MESSAGES': true } );
+				await channel.updateOverwrite( chatModRole, { 'SEND_MESSAGES': true } );
+				await channel.updateOverwrite( superModRole, { 'SEND_MESSAGES': true } );
 
```

Comment:
for better performance this code could could make use of its async property.
e.g. parallel execution where applicable, also please fix inconsistent spacing: 

```suggestion
// send message first, then update channel
await channel.updateOverwrite(message.guild.roles.everyone, {'SEND_MESSAGES': false}, "channel freeze");
await Promise.all([
    channel.updateOverwrite(memberRole, { 'SEND_MESSAGES': false }, "channel freeze"),
    channel.updateOverwrite(chatModRole, { 'SEND_MESSAGES': true }),
    channel.updateOverwrite(superModRole, { 'SEND_MESSAGES': true })
]);
```